src/goto-harness/recursive_initialization.cpp

```diff
@@ -129,3 +134,23 @@ void recursive_initializationt::initialize_nondet(
 {
   body.add(code_assignt{lhs, side_effect_expr_nondett{lhs.type()}});
 }
+
+void recursive_initializationt::initialize_array(
+  const exprt &array,
+  std::size_t depth,
+  const recursion_sett &known_tags,
+  code_blockt &body)
+{
+  PRECONDITION(array.type().id() == ID_array);
+  const auto &array_type = to_array_type(array.type());
+  const auto array_size =
+    numeric_cast_v<std::size_t>(to_constant_expr(array_type.size()));
```

Comment:
As of right now, yes. Atm we only initialise global variables and function parameters, VLAs are not allowed in either of these contexts. Extending this to work with VLAs would be a bit tricky to be honest, because we unroll loops statically. The only reason we do this though is because we ran into a number of bugs trying to make this work with dynamic allocation, once those are fixed we can just use dynamic allocation and leave the unrolling of loops to cbmc, which should make this more flexible and the generated code would become a lot more readable as a bonus.

Reply:
Could you please add a comment to the code, pretty much stating what you just said? It seems perfectly possible that this invariant fails at some point, and then the person doing the debugging should have all the information immediately available.